### PR TITLE
04 get api articles article

### PR DESCRIPTION
**Pull request by seb-the-architect**
Next() is now used for error handing in the catch block.

An empty object is returned when a numerical, non-existant article_id is queried.
A test has been added to prove this.

Thanks kyle. (or whoever checks it this time :)

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -1,27 +1,64 @@
...
 
-//afterAll(() => {return pool.end()})
+afterAll(() => { return pool.end() })
 
-describe("GET /api/topics", () => 
-{
-    test('Every element in the returned array is an object, containing slug and description as properties', () => 
-    {
+describe("GET /api/topics", () => {
+    test('Every element in the returned array is an object, containing slug and description as properties', () => {
```
Comment (MrMcSnail) [inline __tests__/app.test.js:12]: These are good, accurate test descriptions. This will help debugging in future.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -1,27 +1,64 @@
...
+
+    test('Expect empty object when querying numerical, non-existent article_id.', () => {
+        return request(app)
+            .get("/api/articles/1337")
+            .expect(200)
+            .then((response) => {
+                expect(response.body).toStrictEqual({});
+            }
+            )
+    })
```
Comment (Angaca) [inline __tests__/app.test.js]: This should return a 404 - article not found

File: __tests__/app.test.js [hunk 3]
```diff
@@ -1,27 +1,64 @@
...
+
+    test('Incorrect user input is rejected with status 400.', () => {
+        return request(app)
+            .get("/api/articles/spaghettios")
+            .expect(400)
+            .then((response) => {
+                expect(response.body.error).toEqual("Bad request. Is the ID supplied correct?")
```
Comment (MrMcSnail) [inline __tests__/app.test.js:50]: Great, your error message reflects the fact that this could eventually make its way to the end user.

**Review by Angaca — APPROVED** [review]
Code is way better now but please fix that 200 for a valid not existing article id which should error with a 404

File: __tests__/app.test.js (hunk 2 above)
Comment (MrMcSnail) [inline __tests__/app.test.js:63]: great test cases and you've been thorough when thinking of error scenarios

**Review by MrMcSnail — APPROVED** [review]

**Pull request merged**